sky/utils/controller_utils.py

```diff
@@ -1186,35 +1187,103 @@ def _sub_path_join(sub_path: Optional[str], path: str) -> str:
-JOB_MEMORY_MB = 350
-# Monitoring process for service is 1GB. This is based on an old estimation but
-# we keep it here for now.
+# Monitoring process for service is 512MB. This is based on an old
+# estimation but we keep it here for now.
 # TODO(tian): Remeasure this.
-SERVE_MONITORING_MEMORY_MB = 1024
-# The ratio of service controller process to job process. We will treat each
-# service as SERVE_PROC_RATIO job processes.
-SERVE_PROC_RATIO = SERVE_MONITORING_MEMORY_MB / JOB_MEMORY_MB
-# Past 2000 simultaneous jobs, we become unstable.
-# See https://github.com/skypilot-org/skypilot/issues/4649.
-MAX_JOB_LIMIT = 2000
-# Number of ongoing launches launches allowed per CPU, for managed jobs.
-JOB_LAUNCHES_PER_CPU = 4
-# Number of ongoing launches launches allowed per CPU, for services. This is
-# also based on an old estimation, but SKyServe indeed spawn a new process
-# for each launch operation, so it should be slightly more resources demanding
-# than managed jobs.
-SERVE_LAUNCHES_PER_CPU = 2
-# The ratio of service launch to job launch. This is inverted as the parallelism
-# is determined by 1 / LAUNCHES_PER_CPU.
-SERVE_LAUNCH_RATIO = JOB_LAUNCHES_PER_CPU / SERVE_LAUNCHES_PER_CPU
+SERVE_MONITORING_MEMORY_MB = 512
+# The resource consumption ratio of service launch to serve down.
+SERVE_LAUNCH_RATIO = 2.0
 
 # The _RESOURCES_LOCK should be held whenever we are checking the parallelism
 # control or updating the schedule_state of any job or service. Any code that
 # takes this lock must conclude by calling maybe_schedule_next_jobs.
 _RESOURCES_LOCK = '~/.sky/locks/controller_resources.lock'
 
+# keep 2GB reserved after the controllers
+MAXIMUM_CONTROLLER_RESERVED_MEMORY_MB = 2048
+
+# NOTE: In the current implementation, we only consider the memory
+# The ratio of resources consumption for managed jobs and pool/serve.
+# This measures pool_resources / jobs_resources. If 2 GB memory is allocated to
+# jobs, then 2 * POOL_JOBS_RESOURCES_RATIO GB memory is allocated to pool/serve.
+POOL_JOBS_RESOURCES_RATIO = 1
+# Number of ongoing launches launches allowed per worker. Can probably be
+# increased a bit to around 16 but keeping it lower to just to be safe
+LAUNCHES_PER_WORKER = 8
+
+# Based on testing, each worker takes around 200-300MB memory. Keeping it
+# higher to be safe.
+JOB_WORKER_MEMORY_MB = 400
+# this can probably be increased to around 300-400 but keeping it lower to just
+# to be safe
+JOBS_PER_WORKER = 200
+
+
+def _get_total_usable_memory_mb(pool: bool, consolidation_mode: bool) -> float:
+    controller_reserved = float(MAXIMUM_CONTROLLER_RESERVED_MEMORY_MB)
+    if pool:
+        controller_reserved *= (1. + POOL_JOBS_RESOURCES_RATIO)
+    total_memory_mb = (common_utils.get_mem_size_gb() * 1024 -
+                       controller_reserved)
+    if not consolidation_mode:
+        return total_memory_mb
+    config = server_config.compute_server_config(
+        deploy=True, quiet=True, reserved_memory_mb=controller_reserved)
```

Comment:
This `controller_reserved` param means that the computation here and the computation during API server start will be different. The used memory here should presumably affect what the API server is actually planning to use, so the config should match the same config the API server gets.
So we should have this reserved_memory_mb also set during the API server startup, right?